File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

bool compare(int a, int b)
{
    return a > b; 
}
int solution(int k, vector<int> tangerine) {
    int answer = 0;
    int arr[10000001] = {0,};
    
    for (int i = 0; i < tangerine.size(); i++)
        arr[tangerine[i]]++;

    sort(arr, arr + 10000001, compare);
        
    for (int i = 0; i < 10000001; i++)
    {
        answer += arr[i];
        if (answer >= k)
            return i + 1;
    }
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerines) {
        int answer = 0;
        
        Map<Integer, Integer> count = new HashMap<>();
        for (int tangerine : tangerines) {
            count.merge(tangerine, 1, Integer::sum);
        }
        
        List<Integer> list = new ArrayList<>(count.keySet());
        list.sort((a, b) -> count.get(b) - count.get(a));
        
        for (int i : list) {
            k -= count.get(i);
            answer++;
            
            if (k <= 0) break;
        }
        
        return answer;
    }
}